pkg/registry/apis/provisioning/controller/repository.go

```diff
@@ -387,3 +395,36 @@ func (rc *RepositoryController) process(item *queueItem) error {
 	}
 	return nil
 }
+
+// TODO: Find a better home.
+func (rc *RepositoryController) encryptSecrets(ctx context.Context, repo *provisioning.Repository) error {
```

Comment:
wouldn't it this be better in the mutator than in the controller so that the secret doesn't get really saved only the encrypted one?